Tools/autotest/carbonix.py

```diff
@@ -201,6 +207,73 @@ def TestGPSPrearm(index):
             # Restore everything
             self.context_pop()
 
+        def wait_for_engine_temp(condition, threshold, timeout=10):
+            """
+            Waits until a MAVLink message satisfies the given condition.
+
+            Args:
+                master: The MAVLink connection object (e.g., mavutil.mavlink_connection).
+                condition: The condition to check ('greater', 'less', 'equal').
+                threshold: The value to compare the message field against.
+                timeout: Maximum time to wait for the condition (in seconds).
+
+            Returns:
+                The MAVLink message if the condition is satisfied, otherwise None.
+
+            Raises:
+                ValueError: If the condition is invalid.
+            """
+            self.progress(f"Waiting for CHT to be {condition} than {threshold}")
+            tstart = self.get_sim_time()
+            while tstart + timeout > self.get_sim_time():
+                m = self.assert_receive_message('EFI_STATUS', timeout=10)
+                cht = m.cylinder_head_temperature
+                rpm = m.rpm
+                throttle_pos = m.throttle_position
+
+                if condition == 'greater' and cht > threshold:
+                    break
+                if condition == 'lesser' and cht < threshold:
+                    break
+            self.progress(f"CHT = {cht}, RPM = {rpm}, THR = {int(throttle_pos)}% in {self.get_sim_time() - tstart} seconds")
+
+        def TestEngineTelemetry():
+            '''Test the engine telemetry check'''
+
+            # Enable ArduPilot's engine prearm check to use wait_ready_to_arm
+            self.progress('Set BIT_PREARM_DIS:0b000 and reboot SITL')
+            self.set_parameter('BIT_PREARM_DIS', 0b000)
+            self.reboot_sitl()
+            self.change_mode('MANUAL')
+            self.progress('Set Safety Switch OFF')
+            self.set_safetyswitch_off()
+
+            # Check for CHT low on startup
+            self.progress('PreArm check : Disable Arming - CHT is too low')
```

Comment:
I'd get rid of all these progress updates. Most of the functions we call here emit a bunch of stuff already. I'd maybe put them to title a large subsection, but I think these are too much and not clear enough.